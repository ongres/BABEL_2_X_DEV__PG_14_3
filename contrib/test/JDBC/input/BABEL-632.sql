SELECT CAST('\x31' AS bytea);
GO

SELECT CAST('0x31' AS bytea);
GO

-- Test bytea column in a table
CREATE TABLE t_bytea(c1 int, c2 bytea);
GO

INSERT INTO t_bytea (c1, c2) VALUES (1, '\x31');
INSERT INTO t_bytea (c1, c2) VALUES (2, '0x31');
GO

-- Test cast bytea to varbinary
SELECT c1, CAST(c2 AS varbinary(10)) FROM t_bytea;
GO

--Test cast bytea to varchar
SELECT c1, CAST(c2 AS varchar(10)) FROM t_bytea;
GO

DROP TABLE t_bytea;
GO
