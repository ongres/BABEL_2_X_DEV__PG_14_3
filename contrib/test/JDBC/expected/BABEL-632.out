SELECT CAST('\x31' AS bytea);
GO
~~START~~
varbinary
31
~~END~~


SELECT CAST('0x31' AS bytea);
GO
~~START~~
varbinary
30783331
~~END~~


-- Test bytea column in a table
CREATE TABLE t_bytea(c1 int, c2 bytea);
GO

INSERT INTO t_bytea (c1, c2) VALUES (1, '\x31');
INSERT INTO t_bytea (c1, c2) VALUES (2, '0x31');
GO
~~ROW COUNT: 1~~

~~ROW COUNT: 1~~


-- Test cast bytea to varbinary
SELECT c1, CAST(c2 AS varbinary(10)) FROM t_bytea;
GO
~~START~~
int#!#varbinary
1#!#31
2#!#30783331
~~END~~


--Test cast bytea to varchar
SELECT c1, CAST(c2 AS varchar(10)) FROM t_bytea;
GO
~~START~~
int#!#varchar
1#!#\x31
2#!#\x30783331
~~END~~


DROP TABLE t_bytea;
GO
